app/src/main/java/com/internship/move/ui/authentication/login/LoginFragment.kt

```diff
@@ -57,14 +65,17 @@ class LoginFragment : Fragment(R.layout.fragment_login) {
         }
 
         binding.loginBtn.setOnClickListener {
-            CoroutineScope(Dispatchers.Main).launch {
-                binding.loginBtn.setIsLoading(true)
-                delay(Constants.LOADING_DELAY)
-                viewModel.login(
+            viewLifecycleOwner.lifecycleScope.launch {
```

Comment:
If you apply what I mentioned in the viewModel, here in the click listener you'll just launch the api call and in the observer perform navigation or whatever else :D